src/test/java/org/apache/pulsar/io/jcloud/format/JsonFormatTest.java

```diff
@@ -55,6 +59,46 @@ public class JsonFormatTest extends FormatTestBase {
     private static final Logger log = LoggerFactory.getLogger(JsonFormatTest.class);
     private JsonFormat format = new JsonFormat();
 
+    @Test
+    public void testJsonBytesRecordWriter() {
+        List<TestRecord> testRecords = Arrays.asList(
+                new TestRecord("key1", 1, null),
+                new TestRecord("key1", 1, new TestRecord.TestSubRecord("aaa")),
+                new TestRecord("key2", 2, new TestRecord.TestSubRecord("aaa"))
+        );
+
+        try {
+            String json = JSON_MAPPER.get().writeValueAsString(testRecords);
+            sendTypedMessages(jsonBytesTopicName.toString(), SchemaType.BYTES,
+                    Arrays.asList(json.getBytes(), json.getBytes()), Optional.empty(), byte[].class);
+
+            Consumer<Message<GenericRecord>> handle = getJSONMessageConsumer(jsonBytesTopicName);
+            consumerMessages(jsonBytesTopicName.toString(), Schema.AUTO_CONSUME(), handle, 2, 2000);
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
+    }
```

Comment:
It looks like we didn't assert the correctness of the values. It's better to assert the value since we are changing the deserialization logic.